### PR TITLE
[testmerge probably idk] keep em guessing: nerfs and buffs dodge expert at the same time, makes the entire system just a little more performant, adds a baby version of the trait for dodge babies

### PR DESCRIPTION
**Pull request by kittysmooch**
## About The Pull Request

changes a few things about dodge expert:

- eliminates an entire proc that had every mob in the game checking every tick to see if it was wearing armor even if it didn't have dodge expert 🤪 
- rewrites how dodge expert interacts with the dodge calc:
  - nerfs the top end of dodge expert by one measly point, meaning that if you have dodge expert and light armor, you get SPEED * 14 to your dodge calc instead of 15. this isn't a big nerf and you are still likely to see 90% chance pretty regularly.
  - if you have dodge expert and medium armor, you get SPEED * 12 to your dodge calc
  - if you have heavy armor, you get NOTHING
  - this system is binary; any medium armor makes you dodge mediumly and any heavy armor makes you fat roll
- adds dodge adept, which gets SPEED * 12 when wearing light armor
- switches a few classes from expert to adept, mostly based on how strong they should be (bandits get expert) versus how strong they are considering other factors of their class (irregular squires get dodge adept because they're squires, knight champion gets adept because i watched one of them solo 4 squires in a spar with expert) or on class identity (whether theyre john dodgeman or just pretty dexterous)

image

## Why It's Good For The Game

maxresdefault

### CODE REVIEW

**Comment by kittysmooch** [comment]
added a fix for wearing slot items that had ARMOR_CLASS_NONE

**Comment by CormanQuewye** [comment]
I've seen similar PR's like this on other servers and it always a result of people not understanding combat mechanics. Stuns, knockdowns, grabs, pain stuns, and now even feints all disable dodging. Things such as pain nerf its ability (meaning that being hit once can be enough to make dodge inefficient). Furthermore, if you're using a quick weapon dodge receives lessened ability. That's not to also mention that classes with dodge expert tend to be on the lower end of the END spectrum so they can effectively defend less.

In other words, if someone is dodge you can: swap to swift intent and grapple them; jump into them; pull out your dagger and stab them; throwing things at them; feint them. Dodge is *really* not that strong it's people just wanting to two-hand a greatsword and press LMB.

tl;dr the code clean up is cool; keep dodge expert as 15 for light; do not add in dodge adept (why would you ever dodge with dodge adept when you can easily get far better benefits utilizing parry with high defense weapons or a buckler?)

**Comment by LittleEvil9** [comment]
Having finally played a round with dodge adept, I can say that this perk made me FEEL useless, on god, I've never had a more stressing combat heavy round before (so much so that I decided to wait until virtues + statpacks are a thing to not stress myself further). I would have better chances going off using a shield with some minimal time spent on a training dummy than using having this dodge perk. This was tested when I had 12 speed btw.

So dodge is supposed to be an alternative defense to parrying, which most people can't tap into without dodge expert to be viable, unless you have 14+ speed. Dodge users not only uses more stamina per defense but have absolute DOG str stats, so now that they can't dodge as well what are we supposed to do against heavy hitters and especially grappling? It was already *the* ultimate counter to dodging and it will only further make us useless for not having a 90% defence chance.

One of the advantages that dodge used to have back then was that it was the only defense against grapple checks, but that was changed and now you can parry them as well, thats super fine. Another was that dodge couldn't be feinted and they changed it to make it so they could, which can lead to the deadly combo of feint + grab, but alright sure, at least we can attempt to sidestep until the debuff goes away. So now what advantages do dodging get against parrying to warrant these nerfs?

So, to play a dodge user in PVP you have to watch out of all these things: people attempting to jump at you, people being on a straight tile to run into you, quick stance grabbing, your stamina (you use more than parrying), feinting, being put out of position vs haste users. All of these mental gymnastics while achieving terrible defence chances for when you DO get hit. As someone who only enjoys playing dodge builds, these changes will kill any variety along with my joy of playing on the server.

So to recap, on top of all these things. We will have really bad chances of defense chances (anyone below 14 spd), are forced to wear light armor for really bad gains, will be glass cannons since almost no dodge class gets CON, and will lack STR too. So why even play a dodge type character if you can't even dodge? Not only would your char be weaker than everyone else, you can't even tap into the power fantasy of being able to dodge.

tl;dr: perk kills the dodge playstyle for everyone that has the dodge adept perk unless they powergame 14/15 spd, then it becomes USABLE.

**Comment by Lutowski** [comment]
briefly un-tming

**Comment by kittysmooch** [comment]
zero chance i actually get around to willfully opening myself up to novelpost saltrants again so yeah this probably aint ever gettin finished
